Update bulk-create-publish-labels-using-powershell.md (#662)
MicrosoftDocs/OfficeDocs-o365seccomp: SecurityCompliance/bulk-create-publish-labels-using-powershell.md

search.appverid: 
- MOE150
- MET150

> description: "In Office 365, you can use retention labels to implement a retention schedule for your organization. As a record manager or compliance officer, you might have hundreds of retention labels to create and publish. You can do this through the UI in the Security and Compliance Center, but creating retention labels one at a time is time-consuming and inefficient. By using the script and .csv files provided below, you can bulk create and publish retention labels and retention label policies. First you create a list of the retention labels and a list of the retention label policies in Excel, and then you bulk create the retention labels and retention label policies in those lists by using PowerShell. This makes it easier to create and publish at one time, all of the retention labels that your retention schedule requires."

Reviewer: ```suggestion
description: "In Office 365, you can use retention labels to implement a retention schedule for your organization. As a record manager or compliance officer, you might have hundreds of retention labels to create and publish. You can do this through the UI in the Security and Compliance Center, but creating retention labels one at a time is time-consuming and inefficient. By using the script and .csv files provided below, you can bulk create and publish retention labels and retention label policies. First you create a list of the retention labels and a list of the retention label policies in Excel, and then you bulk create the retention labels and retention label policies in those lists by using PowerShell. This makes it easier to create and publish all of the retention labels that your retention schedule requires at the same time."
```

---

# Bulk create and publish retention labels by using PowerShell

In Office 365, you can use retention labels to implement a retention schedule for your organization. As a record manager or compliance officer, you might have hundreds of retention labels to create and publish. You can do this through the UI in the Security &amp; Compliance Center, but creating retention labels one at a time is time-consuming and inefficient.
  

> By using the script and .csv files provided below, you can bulk create retention labels and publish retention label policies. First you create a list of the retention labels and a list of the retention label policies in Excel, and then you bulk create the retention labels and retention label policies in those lists by using PowerShell. This makes it easier to create and publish at one time, all of the retention labels that your retention schedule requires.

Reviewer: ```suggestion
By using the script and .csv files provided below, you can bulk create retention labels and publish retention label policies. First you create a list of the retention labels and a list of the retention label policies in Excel, and then you bulk create the retention labels and retention label policies in those lists by using PowerShell. This makes it easier to create and publish all of the retention labels that your retention schedule requires at the same time.
```

  
For more information about retention labels, see [Overview of labels](labels.md).
  
## Disclaimer

The sample scripts provided in this topic aren't supported under any Microsoft standard support program or service. The sample scripts are provided AS IS without warranty of any kind. Microsoft further disclaims all implied warranties including, without limitation, any implied warranties of merchantability or of fitness for a particular purpose. The entire risk arising out of the use or performance of the sample scripts and documentation remains with you. In no event shall Microsoft, its authors, or anyone else involved in the creation, production, or delivery of the scripts be liable for any damages whatsoever (including, without limitation, damages for loss of business profits, business interruption, loss of business information, or other pecuniary loss) arising out of the use of or inability to use the sample scripts or documentation, even if Microsoft has been advised of the possibility of such damages.
  
## Step 1: Create a .csv file for creating the retention labels

> First you create a .csv file that contains a list of your retention labels with their settings. You can use the sample below as a template by copying it into Excel, converting the text to columns (in Excel \> **Data** tab \> **Text to Columns** \> **Delimited** \> **Comma** \> **General**), and then saving the worksheet as a .csv file in a location that's easy to find.

Reviewer: ```suggestion
First create a .csv file that contains a list of your retention labels with their settings. You can use the sample below as a template by copying it into Excel, converting the text to columns (in Excel \> **Data** tab \> **Text to Columns** \> **Delimited** \> **Comma** \> **General**), and then saving the worksheet as a .csv file in a location that's easy to find.
```

  
For more information about the parameter values for this cmdlet, see [New-ComplianceTag](https://go.microsoft.com/fwlink/?linkid=866511).
  
Notes:
  
- If you don't provide a source file for creating retention labels, the script moves on and prompts you for the source file for publishing retention labels (see the next section), and the script will publish only existing retention labels.
    

> - If the .csv file contains a retention label with the same name as one that already exists, the script skips creating that retention label. No duplicate retention labels are created.

Reviewer: ```suggestion
- If the .csv file contains a retention label with the same name as one that already exists, the script skips creation of that retention label. No duplicate retention labels are created.
```

  
- [Connect to Office 365 Security &amp; Compliance Center PowerShell](https://go.microsoft.com/fwlink/?linkid=799771)
    
## Step 5: Run the PowerShell script to create and publish the retention labels

> After you've connected to Security &amp; Compliance Center PowerShell, next you run the script that creates and publishes the retention labels.

Reviewer: ```suggestion
Connect to Security &amp; Compliance Center PowerShell and run the script that creates and publishes the retention labels.
```

## Step 6: View the log file with the results

> When you run the script, it generates a log file that records each action it took and whether the action succeeded or failed. The log file includes all metadata about the retention labels created and the retention labels published. You can find the log file at this location -- note that the digits in the file name vary.

Reviewer: ```suggestion
When you run the script, it generates a log file that records each action it took and whether the action succeeded or failed. The log file includes all metadata about the retention labels created and the retention labels published. You can find the log file at this location (note that the digits in the file name vary):
```
